src/platform/utilities/api/index.js

```diff
@@ -56,22 +57,29 @@ export function apiRequest(resource, optionalSettings = {}, success, error) {
         ? response.json()
         : Promise.resolve(response);
 
-      if (!response.ok) {
+      if (response.ok || response.status === 304) {
+        const sessionExpiration = response.headers.get('X-Session-Expiration');
+        if (sessionExpiration)
+          localStorage.setItem('sessionExpiration', sessionExpiration);
+        return data;
+      }
+
+      if (environment.isProduction()) {
```

Comment:
Once the session timeout modal launches on production, we will no longer need this function to handle any redirects. The modal will become responsible for knowing when the session expires and redirecting as appropriate.

Reply:
For the scenario where a logged out user attempts to visit a page that makes an authenticated request, the same redirect will occur for any apps that use RequiredLoginView. Ultimately, it doesn't seem like the responsibility of redirecting and clearing the session data belongs to this method.